src/main/java/com/thedariusz/warnme/twitter/TweetService.java

```diff
@@ -56,55 +56,81 @@ public void syncTweets(String twitterUserId) {
+//        TweetType tweetTypeFromTags = getTweetTypeBasedOnHashTags(tweetDto.getHashtagsFromTweet());
+//        TweetType tweetTypeFromText = getTweetTypeBasedOnTweetText(tweetDto.getText());
+//
+//        return tweetTypeFromTags.equals(TweetType.METEO_ALERT) ||
+//                tweetTypeFromText.equals(TweetType.METEO_ALERT);
+//    }
+//
+//    TweetType getTweetTypeBasedOnHashTags(List<Hashtag> hashTags) {
+//        if (isEmpty(hashTags)) {
+//            return TweetType.OTHER;
+//        }
+//
+//        TweetType tweetType = hashTags
+//                .stream()
+//                .map(Hashtag::getTag)
+//                .map(String::toLowerCase)
+//                .anyMatch(getMeteoKeywords()::contains) ? TweetType.METEO : TweetType.OTHER;
+//
+//        boolean hasMeteoAlertKeywords = hashTags
+//                .stream()
+//                .map(Hashtag::getTag)
+//                .map(String::toLowerCase)
+//                .anyMatch(getMeteoAlertKeywords()::contains);
+//
+//        if (tweetType.equals(TweetType.METEO) && hasMeteoAlertKeywords) {
+//            tweetType = TweetType.METEO_ALERT;
+//        }
+//        return tweetType;
+//    }
+//
+//    private TweetType getTweetTypeBasedOnTweetText(String text) {
+//        if (text.isBlank()) {
+//            return TweetType.OTHER;
+//        }
+//
+//        String lowerCaseText = text.toLowerCase();
+//
+//        TweetType tweetType = getMeteoKeywords().stream()
+//                .anyMatch(lowerCaseText::contains) ? TweetType.METEO:TweetType.OTHER;
+//
+//        if (TweetType.METEO.equals(tweetType)) {
+//            tweetType = getMeteoAlertKeywords().stream()
+//                    .anyMatch(lowerCaseText::contains) ? TweetType.METEO_ALERT : tweetType;
+//        }
+//
+//        return tweetType;
+//    }
 
-        return tweetTypeFromTags.equals(TweetType.METEO_ALERT) ||
-                tweetTypeFromText.equals(TweetType.METEO_ALERT);
-    }
+    boolean isMeteoAlert(TweetDto tweetDto) {
+        TweetType tweetType = TweetType.OTHER;
 
-    TweetType getTweetTypeBasedOnHashTags(List<Hashtag> hashTags) {
-        if (isEmpty(hashTags)) {
-            return TweetType.OTHER;
+        if (tweetHashtagsContainsKeywords(tweetDto.getHashtagsFromTweet(), getMeteoKeywords()) ||
+                tweetTextFieldContainsKeywords(tweetDto.getText(), getMeteoKeywords())) {
+            tweetType=TweetType.METEO;
         }
 
-        TweetType tweetType = hashTags
-                .stream()
-                .map(Hashtag::getTag)
-                .map(String::toLowerCase)
-                .anyMatch(getMeteoKeywords()::contains) ? TweetType.METEO : TweetType.OTHER;
+        return tweetType.equals(TweetType.METEO) && (
+                tweetHashtagsContainsKeywords(tweetDto.getHashtagsFromTweet(), getMeteoAlertKeywords()) ||
+                        tweetTextFieldContainsKeywords(tweetDto.getText(), getMeteoAlertKeywords()));
+    }
```

Comment:
Please extract this logic to method where you name this logical operation because i would need at least 5 min to understand when this method returns true or false